app/controllers/application_controller.rb

```diff
@@ -25,8 +25,8 @@ def append_info_to_payload(payload)
     payload[:ip] = request.remote_ip
   end
 
-  def analytics
-    @analytics ||= Analytics.new(current_user, request_attributes, ahoy)
+  def analytics(service = Analytics.new(current_user, request))
+    @analytics ||= service
```

Comment:
Hmm...   this seems a little strange.   What happens if you call analytics() without and argument then later call it with a fake?   Seems like the @ caching and the argument are somewhat at odds with each other.